server/resp/src/main/java/org/infinispan/server/resp/commands/list/blocking/AbstractBlockingPop.java

```diff
@@ -93,6 +94,13 @@ private CompletableFuture<Collection<byte[]>> addSubscriber(PopConfiguration con
          pubSubListener.startTimer(configuration.timeout());
          pubSubListener.synchronizer.onListenerAdded();
       });
+      ClientMetadata metadata = handler.respServer().metadataRepository().client();
+      metadata.incrementBlockedClients();
+      metadata.recordBlockedKeys(configuration.keys().size());
+      pubSubListener.getFuture().thenRun(() -> {
```

Comment:
The decrement happens only if there are no exceptions. Shouldn't we use a `handle()` method instead ?

Reply:
Yeah, let me update it and double-check